docs/pages/react/login-methods/onchain-passkeys.mdx

```diff
@@ -0,0 +1,131 @@
+  adding a secondary off-chain signer to use for account recovery
+</Warning>
+
+```tsx twoslash
+import { createWebAuthnCredential } from "viem/account-abstraction";
+
+const credential = await createWebAuthnCredential({
+  name: "Credential Name",
+});
+
+// store credential id to public key mapping
+// NOTE: use of localStorage is for TESTING PURPOSES ONLY, NOT FOR PRODUCTION USE
+localStorage.setItem(credential.id, credential.publicKey); //credentialIdAsBase64Encoded -> publicKeyHex
+```
+
+### Login With Credential
+
+```tsx twoslash
+import { createModularAccountV2Client } from "@account-kit/smart-contracts";
+import { alchemy, arbitrumSepolia } from "@account-kit/infra";
+
+const publicKeyRequest: PublicKeyCredentialRequestOptions = {
+  challenge: Uint8Array.fromHex("0x"), // Generate a random challenge
+  rpId: "localhost", // should match your dApp domain
+};
+
+// retrieve available passkeys for the provided domain
+const publicKeyCredential = await navigator.credentials.get({
+  publicKeyRequest,
+});
+
+if (publicKeyCredential) {
+  // verify that passkey with corresponding id exists on dApp
+  const publicKeyHex = localStorage.getItem(publicKeyCredential.id);
+  if (!publicKeyHex) throw new Error("Account does not exist");
+
+  // create client to send transactions on behalf of verified user
+  const accountClient = await createModularAccountV2Client({
+    policyId: "YOUR_POLICY_ID",
+    mode: "webauthn",
+    credential: {
+      id: publicKeyCredential.id,
+      publicKey: publicKeyHex,
+    },
+    rpId: "localhost",
+    chain: arbitrumSepolia,
+    transport: alchemy({ apiKey: "YOUR_ALCHEMY_API_KEY" }),
+  });
+}
+```
+
+<Accordion title="`createModularAccountV2Client` Parameters">
+
+| Parameter    | Type         | Description                                         |
+| ------------ | ------------ | --------------------------------------------------- |
+| `policyId`   | `string`     | Your account policy UUID from the Alchemy dashboard |
+| `mode`       | `"webauthn"` | Specifies credential mode                           |
+| `credential` | `object`     | `{ id: string, publicKey: Address }`                |
+| `rpId`       | `string`     | Relying Party ID (e.g., `localhost` or your domain) |
+| `chain`      | `Chain`      | Network config (e.g., `arbitrumSepolia`)            |
+| `transport`  | `Transport`  | Alchemy or custom RPC transport                     |
+
+</Accordion>
+
+### Send User Operation
+
+```tsx
+const operation = await accountClient.sendUserOperation({
+  uo: {
+    target: "0xd8dA6BF26964aF9D7eEd9e03E53415D37aA96045", // Example: Vitalik's address
+    data: "0x", // No calldata
+    value: parseEther("0"),
+  },
+});
+```
+
+## React Native Integration
+
+Get your React Native environment set up by following these [docs](/wallets/react-native/overview). Once you’ve completed this setup, you can use the webauthn signer as detailed above!
```

Comment:
should we call out that react native doesn't have `localstorage` (since that's used in the example above), or is it fine as-is since that will prob be obvious to react native devs?

Reply:
good callout! it's probably better to err on the side of caution, I'll add a note about this